tools/kicad/kicad_rs/src/eval/display.rs

```diff
@@ -0,0 +1,109 @@
+use evalexpr::Value;
+use std::fmt::{Display, Formatter, Result};
+
+const UNITS_LARGE: &[char] = &['k', 'M', 'G', 'T', 'E'];
+const UNITS_SMALL: &[char] = &['m', 'µ', 'n', 'p'];
+
+pub struct PrettyPrintValue<'a> {
+    value: &'a Value,
+    unit: &'a Option<String>,
+}
+
+impl<'a> PrettyPrintValue<'a> {
+    pub fn new(value: &'a Value, unit: &'a Option<String>) -> Self {
+        Self { value, unit }
+    }
+}
+
+impl<'a> Display for PrettyPrintValue<'a> {
+    fn fmt(&self, f: &mut Formatter) -> Result {
+        match self.value {
+            Value::Float(mut float) => {
+                let mut prefix = None;
+                if let Some((mut iter, mul)) = match float {
+                    _ if float == 0.0 => None,
+                    _ if float.abs() > 1000.0 => Some((UNITS_LARGE.iter(), 0.001)),
+                    _ if float.abs() < 1.0 => Some((UNITS_SMALL.iter(), 1000.0)),
+                    _ => None,
+                } {
+                    while let (i @ Some(_), true) =
+                        (iter.next(), float.abs() > 1000.0 || float.abs() < 1.0)
+                    {
+                        prefix = i;
+                        float *= mul;
+                    }
+                }
+
+                write!(f, "{:.2}", float)?; // Fixed two-decimal precision
+                if self.unit.is_some() {
+                    write!(f, " ")?;
+                }
+
+                prefix.map(|u| write!(f, "{}", u)).transpose()?;
+                self.unit.as_ref().map(|u| write!(f, "{}", u)).transpose()?;
+
+                Ok(())
+            }
+            other => {
+                other.fmt(f)?;
+                if let Some(unit) = self.unit {
+                    write!(f, " {}", unit)?;
+                }
+                Ok(())
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    macro_rules! test_map {
```

Comment:
This seems also useful for unit testing other modules. Should this be extracted to a new file and imported instead?

Reply:
I've renamed it to `test_ppv` now (forgot to do that earlier), it's specific to testing the `PrettyPrintValue` so I won't move it out right now. We can come up with a more generic approach later if needed.